### PR TITLE
stm32: Add support for B-L072Z-LRWAN1 LoRa modem

### PR DESCRIPTION
**Pull request by projectgus**
* Adds pin definitions for the internal module modem connections.
* One-line fix for `MICROPY_HW_SPI1_SCK` pin.
* Lower the default feature level to "Core" in order to have more room for frozen modules, etc. (Increases free space significantly.)

See commit messages for more details.

With this PR and the latest https://github.com/micropython/micropython-lib/pull/501 then it's possible to use the LoRa modem:

```py
from machine import Pin, SPI
# SX_SCK, SX_MOSI, SX_MISO are the default SPI1 pins  
modem = SX1276(
    spi=SPI(1, baudrate=2000_000, polarity=0, phase=0),
    cs = Pin.board.SX_CS,
    dio0=Pin.board.SX_DIO0,
    dio1=Pin.board.SX_DIO1,
    reset=Pin.board.SX_RESET,
    lora_cfg=lora_cfg,
)
```

(Can substitute the above into the `simple_rxtx` example from the LoRa driver PR. The `reliable_delivery` example wasn't tested as it requires more MicroPython features.)

Once the micropython-lib LoRa PR is reviewed then I plan to submit some board/module wrapper packages so it'll be possible to install a board-specific package and then do `from lora import L072Z_LRWAN1` or similar.

### CODE REVIEW

**Comment by projectgus** [comment]
No uasyncio support in this PR. With the amount of free flash then it may be possible to enable it and use the async lora driver, but I think the remaining free resources (flash, RAM) will end up pretty low all around.

File: ports/stm32/boards/B_L072Z_LRWAN1/mpconfigboard.h [hunk 1]
```diff
@@ -5,6 +5,11 @@
 #define MICROPY_HW_BOARD_NAME       "B-L072Z-LRWAN1"
 #define MICROPY_HW_MCU_NAME         "STM32L072CZ"
 
+#define MICROPY_CONFIG_ROM_LEVEL (MICROPY_CONFIG_ROM_LEVEL_CORE_FEATURES)
+
+#define MICROPY_HELPER_REPL (1)
+#define MICROPY_KBD_EXCEPTION (1)
+#define MICROPY_ENABLE_SCHEDULER (1)
```
Comment (dpgeorge) [inline ports/stm32/boards/B_L072Z_LRWAN1/mpconfigboard.h:12]: indent to match other lines

File: ports/stm32/boards/B_L072Z_LRWAN1/mpconfigboard.h [hunk 2]
```diff
@@ -41,8 +47,8 @@
 #define MICROPY_HW_I2C1_SDA         (pin_B9)
 
 // SPI buses
-#define MICROPY_HW_SPI1_NSS         (pin_A15)
-#define MICROPY_HW_SPI1_SCK         (pin_A5)
+#define MICROPY_HW_SPI1_NSS         (pin_A15) /* SPI1 on the CMWX1ZZABZ module is connected internally to the SX1276 */
```
Comment (dpgeorge) [inline ports/stm32/boards/B_L072Z_LRWAN1/mpconfigboard.h:50]: C++ style comments

**Comment by projectgus** [comment]
@dpgeorge Rebased, updated according to review comments & discussion. Thanks!

**Comment by dpgeorge** [comment]
Thanks for updating.  This looks good to merge now.

**Comment by mattytrentini** [comment]
Would it be good idea to create a small `stm32/boards/B_L072Z_LRWAN1/README.md` to provide information how to set up LoRa on this board? As in, little more than `mip.install("lora-sync")` and that snippet of code that @projectgus provided at the top? 

Since the purpose of the board is clearly to use LoRa, it seems like it would be good to help folks get LoRa working. :) I can raise the PR if it's a good idea...

**Comment by projectgus** [comment]
@mattytrentini Excellent point! There are some ideas about adding micropython-lib packages for some of these small bits of setup code to make it even easier, I'm planning to write up an RFC soon for discussion.

Until then, though, putting this in a README instead of the PR description makes a ton of sense. If you've got time to put up a PR for it then I've got time to test it (unless you also have an L072Z board to test). Otherwise I don't mind writing it out myself.

**Comment by mattytrentini** [comment]
I've added a brief README in #12063 but I don't have that board - so if you could please test it @projectgus I'd much appreciate it! And feel free to make any other suggestions for additions or changes to the README.

**Pull request merged**